File: easy/3345_smallest_divisible_digit_product_i/solution.cpp
class Solution {
public:
  int smallestNumber(int n, int t) {
    while (true) {
      int nCopy = n++;
      int digitsProduct = 1;

      while (nCopy) {
        digitsProduct *= nCopy % 10;
        nCopy /= 10;
      }

      if (digitsProduct % t == 0)
        return n - 1;
    }

    return -1;
  }
};

File: easy/3345_smallest_divisible_digit_product_i/solution.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        while True:
            nCopy = n
            digitsProduct = 1

            while nCopy:
                digitsProduct *= nCopy % 10
                nCopy //= 10

            if digitsProduct % t == 0:
                return n
            else:
                n += 1

        return -1
